apps/frontend/src/components/xpress/XpressProposalTable.tsx

```diff
@@ -0,0 +1,389 @@
+          });
+
+        if (result.proposals === undefined) {
+          return;
+        }
+        const tableData =
+          result.proposals.map((proposal) => {
+            const selection = new Set(searchParams.getAll('selection'));
+            const proposalData = {
+              ...proposal,
+              status: proposal.submitted ? 'Submitted' : 'Open',
+              technicalReviews: proposal.technicalReviews?.map(
+                (technicalReview) => ({
+                  ...technicalReview,
+                  status: getTranslation(technicalReview.status as ResourceId),
+                })
+              ),
+              finalStatus: getTranslation(proposal.finalStatus as ResourceId),
+            } as ProposalViewData;
+
+            if (searchParams.getAll('selection').length > 0) {
+              return {
+                ...proposalData,
+                tableData: {
+                  checked: selection.has(proposal.primaryKey.toString()),
+                },
+              };
+            } else {
+              return proposalData;
+            }
+          }) || [];
+
+        setTableData(tableData);
+
+        resolve({
+          data: tableData,
+          page: tableQuery.page,
+          totalCount: result.totalCount,
+        });
+      } catch (error) {
+        reject(error);
+      }
+    });
+
+  const handleSortOrderChange = (orderByCollection: OrderByCollection[]) => {
+    const [orderBy] = orderByCollection;
+    setSearchParams((searchParam) => {
+      searchParam.delete('sortField');
+      searchParam.delete('sortDirection');
+      if (orderBy?.orderByField != null && orderBy?.orderDirection != null) {
+        searchParam.set('sortField', orderBy?.orderByField);
+        searchParam.set('sortDirection', orderBy?.orderDirection);
+      }
+
+      return searchParam;
+    });
+  };
+
+  const handleFilterChange = (filter: ProposalsFilter) => {
+    setProposalFilter(filter);
+    refreshTableData();
+  };
+
+  const { calls, loadingCalls } = useCallsData();
+
+  const { instruments, loadingInstruments } = useXpressInstrumentsData(
+    tableData,
+    techniques,
+    calls
+  );
+
+  const handleSearchChange = (searchText: string) => {
+    setSearchParams({
+      search: searchText ? searchText : '',
+      page: searchText ? '0' : page || '',
+    });
+  };
+
+  return (
+    <>
+      <StyledContainer maxWidth={false}>
+        <StyledPaper data-cy="xpress-proposals-table">
+          <XpressProposalFilterBar
+            calls={{ data: calls, isLoading: loadingCalls }}
+            instruments={{ data: instruments, isLoading: loadingInstruments }}
+            techniques={{
+              data: techniques,
+              isLoading: loadingTechniques,
+            }}
+            proposalStatuses={{
+              data: proposalStatuses,
+              isLoading: loadingProposalStatuses,
+            }}
+            handleFilterChange={handleFilterChange}
+            filter={proposalFilter}
```

Comment:
We can also move filter into to handleFilterChange function and not pass it if possible since it is not being used for any computations in XpressProposalFilterBar. Looks good just those last comments.

Reply:
@mutambaraf, sorry I didn't understand it properly. 'filter' holds the currently selected values of each filters right. We would need it to know the current filter values inside XpressProposalFilterBar.